client/ayon_core/modules/deadline/plugins/publish/collect_user_credentials.py

```diff
@@ -3,32 +3,85 @@
 
 Requires:
     context -> project_settings
+    instance.data["deadline"]["url"]
+        or instance.data["deadlineUrl"] for backward compatibility, remove soon
 
 Provides:
-    context -> deadline_require_authentication (bool)
-    context -> deadline_auth (tuple (str, str)) - (username, password) or None
+    instance.data["deadline"] -> require_authentication (bool)
+    instance.data["deadline"] -> auth (tuple (str, str)) -
+        (username, password) or None
 """
 import pyblish.api
 
+from ayon_api import get_server_api_connection
+from ayon_core.modules.deadline.deadline_module import DeadlineModule
+from ayon_core.modules.deadline import __version__
 
-class CollectDeadlineUserCredentials(pyblish.api.ContextPlugin):
+
+class CollectDeadlineUserCredentials(pyblish.api.InstancePlugin):
     """Collects user name and password for artist if DL requires authentication
     """
 
     # Run before collect_deadline_server_instance.
-    order = pyblish.api.CollectorOrder
+    order = pyblish.api.CollectorOrder + 0.200
     label = "Collect Deadline User Credentials"
 
-    def process(self, context):
-        deadline_settings = context.data["project_settings"]["deadline"]
+    hosts = ["aftereffects",
+             "fusion",
+             "harmony"
+             "nuke",
+             "maya",
+             "max",
+             "houdini"]
+
+    families = ["render",
+                "rendering",
+                "render.farm",
+                "renderFarm",
+                "renderlayer",
+                "maxrender",
+                "usdrender",
+                "redshift_rop",
+                "arnold_rop",
+                "mantra_rop",
+                "karma_rop",
+                "vray_rop",
+                "publish.hou"]
+
+    def process(self, instance):
+        # backward compatibility, remove soon
```

Comment:
Can we mark this `# todo:` - they should be highlighted by most IDEs then and at least it's clear the "remove soon" has the intent of something to yet be done.

Reply:
Removed altogether, was on the fence about it, but it shouldn't actually need to be there.